staging/kos/pkg/apis/network/types.go

```diff
@@ -161,6 +188,50 @@ func Now() metav1.MicroTime {
 	return metav1.NowMicro()
 }
 
+// ClientWrite models a write by a client. A "client" is any entity that is not
+// part of the KOS control plane.
+type ClientWrite struct {
+	// Name identifies the client write.
+	Name string
+
+	// The time at which the client write happened.
+	Time metav1.MicroTime
+}
+
+// IsValid checks whether `cw` fields are consistent.
+func (cw ClientWrite) IsValid() bool {
+	return timedEventIsValid(cw.Name, cw.Time)
+}
+
+// ControllerStart carries information on the start of a KOS controller.
+type ControllerStart struct {
+	// Controller is the name of the controller which started.
+	Controller string
+
+	// ControllerTime is the time at which the controller started, as recorded
+	// by the controller itself.
+	ControllerTime metav1.MicroTime
+}
+
+// IsValid checks whether `cs` fields are consistent.
+func (cs ControllerStart) IsValid() bool {
+	return timedEventIsValid(cs.Controller, cs.ControllerTime)
+}
+
+func timedEventIsValid(event string, time metav1.MicroTime) bool {
+	return (event == "" && time == metav1.MicroTime{}) ||
+		(event != "" && time != metav1.MicroTime{})
+}
+
+const (
+	SubnetClientWrite = "subnet"
+	NAClientWrite     = "na"
+
+	SVControllerStart   = "subnet_validator"
```

Comment:
Fixed in 5c55ad8 .

We probably want to do the same thing to the consts for ClientWrites (right above, lines 227-228), but we have to find a workaround since simply dropping "ClientWrite" causes naming collisions.